server/src/main/java/org/eclipse/openvsx/storage/IStorageService.java

```diff
@@ -30,9 +32,14 @@ public interface IStorageService {
      */
     void removeFile(FileResource resource);
 
+    /**
+     * Returns the cache control duration of a resource.
+     */
+    Duration getCacheDuration(FileResource resource);
+
```

Comment:
Can you remove this method from the interface?

Instead, let's create a `@Component` in the storage package called `FileCacheDurationConfig` with a `@Value(ovsx.storage.file-cache-duration:P7D) String cacheDuration`.
The value can be converted to a `Duration` by calling [Duration.parse();](https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html#parse-java.lang.CharSequence-)

You can then use `FileCacheDurationConfig` at https://github.com/eclipse/openvsx/blob/acbee4f8c3e71ba43ae81bc6300b0ea35d7040b0/server/src/main/java/org/eclipse/openvsx/LocalRegistryService.java#L174

and at
https://github.com/eclipse/openvsx/blob/acbee4f8c3e71ba43ae81bc6300b0ea35d7040b0/server/src/main/java/org/eclipse/openvsx/storage/AwsStorageService.java#L110-L114

Please use the newer `java.time` API here to convert the `FileCacheDurationConfig.cacheDuration`  `Duration` to a `Date` object:
```
instant = localDateTime.toInstant(ZoneOffset.UTC);
date = Date.from(instant);
```

Reply:
The reason I had implemented `Duration getCacheDuration(FileResource resource);` was so I could change the cache duration of the `.vsix` in particular, I was planning to cache the media / other files for the original 7 days. If you do not think this is a good idea, I will remove it all in favor of the `FileCacheDurationConfig` component